includes/abstracts/llms-abstract-query.php

```diff
@@ -0,0 +1,589 @@
+<?php
+/**
+ * LLMS_Abstract_Query class file
+ *
+ * @package LifterLMS/Abstracts/Classes
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Database Query abstract class.
+ *
+ * @since [version]
+ *
+ * Query Arguments
+ *
+ * @param int $page             Results page number. Default: `1`.
+ * @param int $per_page         Number of results to retrieve per page. Default: `10`.
+ * @param int $search           Text search term or query. Default: `''`.
+ * @param int $sort             Result order and orderby. Default: `array()`.
+ * @param int $suppress_filters Whether or not to allow filter hooks to run. Default: `false`.
+ * @param int $no_found_rows    Whether or not to disable the total number of found results for the query. Default: `false`.
+ */
+abstract class LLMS_Abstract_Query {
+
+	/**
+	 * Identify the extending query.
+	 *
+	 * @var string
+	 */
+	protected $id = '';
+
+	/**
+	 * Defines fields that can be sorted on via ORDER BY.
+	 *
+	 * If this is not defined by extending classes, the sort data
+	 * will not be validated.
+	 *
+	 * @var string[]|null
+	 */
+	protected $allowed_sort_fields = array();
+
+	/**
+	 * Combined arguments prior to sanitization.
+	 *
+	 * This is the final resulting arguments used to generate the query,
+	 * the result of the default arguments merged into the submitted arguments.
+	 *
+	 * @var array
+	 */
+	protected $arguments = array();
+
+	/**
+	 * Default arguments before merging with original.
+	 *
+	 * @var  array
+	 */
+	protected $arguments_default = array();
+
+	/**
+	 * Original arguments before merging with defaults.
+	 *
+	 * @var array
+	 */
+	protected $arguments_original = array();
+
+	/**
+	 * Total number of results matching query parameters.
+	 *
+	 * @var integer
+	 */
+	protected $found_results = 0;
+
+	/**
+	 * Maximum number of pages of results based off per_page & found_results.
+	 *
+	 * @var integer
+	 */
+	protected $max_pages = 0;
+
+	/**
+	 * Number of results on the current page.
+	 *
+	 * @var integer
+	 */
+	protected $number_results = 0;
+
+	/**
+	 * The final query used to retrieve results.
+	 *
+	 * For a raw database query, this is the SQL passed to `$wpdb->get_results()`. For a WP_Posts query
+	 * this is the array of query arguments passed into the `WP_Query`. Other extending queries
+	 * may used this as they see fit.
+	 *
+	 * @var mixed
+	 */
+	protected $query = array();
+
+	/**
+	 * The parsed and sanitized arguments ultimately used by the query.
+	 *
+	 * @var array
+	 */
+	protected $query_vars = array();
+
+	/**
+	 * Array of results retrieved by the query.
+	 *
+	 * @var array
+	 */
+	protected $results = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.8.0
+	 *
+	 * @param array $args Optional. Query arguments. Default empty array.
+	 *                    When not provided the default arguments will be used.
```

Comment:
```suggestion
	 * @param array $args Query arguments. When not provided the default arguments will be used.
```
We decided to drop this optional thing right?